tests/chainer_tests/test_reporter.py

```diff
@@ -216,5 +217,208 @@ def test_float(self):
         testing.assert_allclose(mean, 2.)
         testing.assert_allclose(std, numpy.sqrt(2. / 3.))
 
+    def test_serialize(self):
+        self.summary.add(1.)
+        self.summary.add(2.)
+
+        summary = chainer.reporter.Summary()
+        testing.save_and_load_npz(self.summary, summary)
+        summary.add(3.)
+
+        mean = summary.compute_mean()
+        testing.assert_allclose(mean, 2.)
+
+        mean, std = summary.make_statistics()
+        testing.assert_allclose(mean, 2.)
+        testing.assert_allclose(std, numpy.sqrt(2. / 3.))
+
+    @attr.gpu
+    def test_serialize_cupy(self):
+        xp = cuda.cupy
+        self.summary.add(xp.array(1, 'f'))
+        self.summary.add(xp.array(2, 'f'))
+
+        summary = chainer.reporter.Summary()
+        testing.save_and_load_npz(self.summary, summary)
+        summary.add(xp.array(3, 'f'))
+
+        mean = summary.compute_mean()
+        testing.assert_allclose(mean, 2.)
+
+        mean, std = summary.make_statistics()
+        testing.assert_allclose(mean, 2.)
+        testing.assert_allclose(std, numpy.sqrt(2. / 3.))
+
+    def test_serialize_backward_compat(self):
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            # old version does not save anything
+            numpy.savez(f, dummy=0)
+            chainer.serializers.load_npz(f.name, self.summary)
```

Comment:
I noticed that warnigns are not catched.
Please use `with testing.assert_warns(UserWarning):` where warning is expected.
e.g. https://github.com/chainer/chainer/blob/v4.0.0b2/tests/chainer_tests/test_cuda.py#L116-L117